authorization/role/service/role_management_service_blackbox_test.go

```diff
@@ -238,3 +251,130 @@ func (s *roleManagementServiceBlackboxTest) compare(createdRole identityrole.Ide
+
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), role)
+
+		scope, err := testsupport.CreateTestScope(s.Ctx, s.DB, *testResourceTypeRef, uuid.NewV4().String())
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), scope)
+
+		rs, err := testsupport.CreateTestRoleScope(s.Ctx, s.DB, *scope, *role)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), rs)
+
+		createdRoleScopes = append(createdRoleScopes, *rs)
+	}
+
+	someOtherResourceType, err := testsupport.CreateTestResourceType(s.Ctx, s.DB, uuid.NewV4().String())
+	require.NoError(s.T(), err)
+
+	for i := 0; i < 3; i++ {
+		role, err := testsupport.CreateTestRole(s.Ctx, s.DB, *someOtherResourceType, uuid.NewV4().String())
+
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), role)
+
+		scope, err := testsupport.CreateTestScope(s.Ctx, s.DB, *someOtherResourceType, uuid.NewV4().String())
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), scope)
+
+		rs, err := testsupport.CreateTestRoleScope(s.Ctx, s.DB, *scope, *role)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), rs)
+	}
+
+	roleScopesRetrieved, err := s.roleManagementService.ListAvailableRolesByResourceType(s.Ctx, testResourceTypeRef.Name)
+	require.NoError(s.T(), err)
+	require.Len(s.T(), roleScopesRetrieved, 10)
+	s.checkRoleBelongsToResourceType(s.DB, roleScopesRetrieved, *testResourceTypeRef)
+	s.checkIfCreatedRoleScopesAreReturned(s.DB, createdRoleScopes, roleScopesRetrieved)
+
+}
+
+func (s *roleManagementServiceBlackboxTest) checkIfCreatedRoleScopesAreReturned(db *gorm.DB, createdRoleScopes []scoperepo.RoleScope, roleScopesRetrieved []rolemodel.RoleScope) {
+	foundCreatedRoleScope := false
+	for _, rsDB := range createdRoleScopes {
+		foundCreatedRoleScope = false
+		for _, rsRetrieved := range roleScopesRetrieved {
+			if rsDB.RoleID.String() == rsRetrieved.RoleID {
+				for _, sc := range rsRetrieved.Scopes {
+					if sc == rsDB.ResourceTypeScope.Name {
+						foundCreatedRoleScope = true
+					}
+				}
+			}
+		}
+	}
+	require.True(s.T(), foundCreatedRoleScope)
+}
+
+func (s *roleManagementServiceBlackboxTest) checkRoleBelongsToResourceType(db *gorm.DB, roleScopesRetrieved []rolemodel.RoleScope, rt resourcetype.ResourceType) {
+	require.True(s.T(), len(roleScopesRetrieved) >= 1)
+	for _, r := range roleScopesRetrieved {
+		roleID, err := uuid.FromString(r.RoleID)
+		require.Nil(s.T(), err)
+
+		existingRole, err := s.roleRepo.Load(s.Ctx, roleID)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), existingRole)
+
+		// this role should belong to the specific resource type
+		require.Equal(s.T(), rt.ResourceTypeID, existingRole.ResourceTypeID)
+		for _, sc := range r.Scopes {
+			foundScope, err := s.checkScopeBelongsToResourceType(s.DB, sc, rt)
+			require.NoError(s.T(), err)
+			require.Equal(s.T(), true, foundScope)
+		}
+	}
+}
+
+func (s *roleManagementServiceBlackboxTest) checkScopeBelongsToResourceType(db *gorm.DB, scopeName string, rt resourcetype.ResourceType) (bool, error) {
```

Comment:
Querying all the resource type scopes just by name, and then iterating through them all to check if one is of a certain resource type seems terribly inefficient, why don't we simply have a Lookup() method in the repository that can return a ResourceTypeScope object for a specified resource type and scope name?